utilities/transactions/pessimistic_transaction.h

```diff
@@ -111,6 +110,12 @@ class PessimisticTxn : public TransactionBaseImpl {
   int64_t GetDeadlockDetectDepth() const { return deadlock_detect_depth_; }
 
  protected:
+  virtual Status PrepareInternal() = 0;
+
+  virtual Status CommitSingleInternal() = 0;
```

Comment:
commit_single is inherited from the existing code. Admitted that it is not a descriptive name. Let me change it to commit_without_prepare and comment to clarify that.

That is a good point. It makes sense to reject commit if 2pc is set. To do that properly we would need to ask users to explicitly set/unset an option in each transaction if they mean to bypass prepare for that. Let us keep that in mind as a future work.